site/content/user-guide/java_client_sender.md

```diff
@@ -0,0 +1,268 @@
+
+### Fields
+
+| Field                                     | Description                                                      |
+| :---------------------------------------- | :--------------------------------------------------------------- |
+| `vertx`  | Hono client is based on vertx, so for a java implementation vertx is the technology to use. Please refer to the vertx documentation for any introduction to that. |
+| `honoUpstreamClient`  | The instance of the hono java client (that is using vertx itself in the implementation)
+| `latch`  | The CountDownLatch is explained below with the code using it.
+
+
+### Constructor
+
+The HonoClient is constructed by a factory builder and is filled with the contact details in a fluent api way.
+The CountDownLatch is initialized for one count down that is used in callbacks to signal the main thread to continue.
+
+### Main method
+
+Initiates the control flow in the method sendUpstreamTelemetryData.
+
+
+### Send upstream telemetry data
+
+This is the most important part of the example and consists of:
+
+
+#### Buildup of a telemetry sender to hono
+
+The buildup of a hono telemetry sender is a typical piece of asynchronous vertx code.
+
+Therefore it is important to not let the main thread end before all AMQP messages were sent and acknowledged - this is were the CountDownLatch is used to block the main thread at the appropriate lines.
+The "latch.await()" code is blocking the main thread until the HonoClient was successfully connected and then starts to send the first message.
+
+
+#### Use the sender to send several messages in sequence to hono
+The message sender instance is constructed in the senderFuture and can be accessed via the result-method.
+
+Here the flow control of the AMQP protocol is relevant: the hono client considers the available credits and invokes the callback once as soon as there are enough credits to send. 
+At the time the callback is invoked the message was sent already.
+
+The callback itself counts down a latch ("messageSenderLatch") which has blocked the main thread. If the latch was counted down, the main thread continues and sends the next message.
+
+
+### Insert logger
+
+You probably noticed the error message at startup that complains about a missing logger binding. You can overcome this by adding a logger to your pom.xml, e.g. logback:
+
+```xml
+<dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <version>1.1.3</version>
+</dependency>
+```
+
+You should get several logging messages during startup then.
+
+## Start and test your implementation
+
+Before you start your application, you normally want to start a receiver for the tenant first. 
+You can use the example receiver of the hono project for that (please see the Getting started page for instructions) or write an own receiver.
+How to write an own java based receiver is described in a similar page to this one in the user guide.
```

Comment:
either include the link to that page or remove this sentence (until you add the page describing the consumer)